kani-compiler/src/codegen_cprover_gotoc/codegen/rvalue.rs

```diff
@@ -1260,26 +1266,13 @@ impl<'tcx> GotocCtx<'tcx> {
 /// Perform a wrapping subtraction of an Expr with a constant "expr - constant"
 /// where "-" is wrapping subtraction, i.e., the result should be interpreted as
 /// an unsigned value (2's complement).
-fn wrapping_sub(expr: &Expr, constant: u128) -> Expr {
-    // While the wrapping subtraction can be done through a regular subtraction
-    // and then casting the result to an unsigned, doing so may result in CBMC
-    // flagging the operation with a signed to unsigned overflow failure (see
-    // https://github.com/model-checking/kani/issues/356).
-    // To avoid those overflow failures, the wrapping subtraction operation is
-    // computed as:
-    //   if expr >= constant {
-    //       // result is positive, so overflow may not occur
-    //       expr - constant
-    //   } else {
-    //       // compute the 2's complement to avoid overflow
-    //       expr - constant + 2^32
-    //   }
-    let s64 = Type::signed_int(64);
-    let expr = expr.clone().cast_to(s64.clone());
-    let twos_complement: i64 = u32::MAX as i64 + 1 - i64::try_from(constant).unwrap();
-    let constant = Expr::int_constant(constant, s64.clone());
-    expr.clone().ge(constant.clone()).ternary(
-        expr.clone().sub(constant),
-        expr.plus(Expr::int_constant(twos_complement, s64.clone())),
-    )
+fn wrapping_sub(expr: &Expr, constant: u64) -> Expr {
+    if constant == 0 {
+        // No need to subtract.
+        expr.clone()
+    } else {
+        let unsigned = Type::unsigned_int(expr.typ().width().unwrap());
```

Comment:
Make this a helper function on the type?

Reply:
Also, this will turn `ssize_t` into `uint64_t` instead of `size_t`.  It would be nice to handle the `CInt` types specially here

Reply:
ok... let me create a helper function in the bindings crate to deal with different types.